pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -437,11 +442,21 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	// TODO(dkistner) Clarify: I guess this work only as we assume the default labels here.
+	// What if different labels are provided via the gardenlet config for the default ingress gateway?
```

Comment:
What are the steps to resolve this TODO?

Reply:
This relate to the topic above. I guess we can make an assumption about the selectors here, but in case the default ingress gateway is configured differently as the default it will not work anymore. This can for instance be done via the GardenletConfiguration in `.sni.ingress.labels["istio"] = "whatever"`.

cc @ScheererJ

Reply:
You are absolutely correct that this is based on the current defaults. @mvladev suggested to use them for now. He wanted to rework the istio handling anyway (back when we discussed it in the context of reversed vpn).

@mvladev please comment.

Reply:
I will check if I can use the default values if I anyway rework the defaulting now a bit.